### DISCUSSION TITLE
Git-LFS-Authenticate command not found -- HELP

### DISCUSSION THREAD

**Discussion by taholmes160** [git-lfs/git-lfs#5763] [General]
HI folks:

The other day I had to uninstall and then re-install git on my windows computer (the software drive was failing, and had to be pulled, so uninstalled everything and then RE-installed it on another drive) ever since, I cant push anything that needs LFS without getting this error:
```
PS L:\l\Current Projects\Vivos> git push origin main
batch request: bash: line 1: git-lfs-authenticate: command not found: exit status 127
error: failed to push some refs to 'ssh://192.168.1.28/mnt/data1/git/vivos.git'
```
there was a similar issue that popped up back in version 2 in 2017, but we are WAY past that now -- so im kinda lost -- can someone please help me resolve this issue?

Thanks
TIM

**Comment by bk2204** [comment]
Hey,

The `git-lfs-authenticate` binary is not provided with Git LFS and is typically implemented by a forge.  That command is designed to provide authentication over SSH for a request over HTTP, so both protocols have to be integrated.

Depending on your Git LFS version (which you haven't specified here), you can also use `git-lfs-transfer` on the remote system to push using the pure SSH protocol.  That is available using the `scutiger-lfs` Rust crate, which you can install using `cargo install` on the remote system.

If you _are_ using a forge of some sort, then you need to configure it to use the right location by using `git remote set-url origin NEW-URL`.

If you weren't intending to push to the local network, then you probably want to figure out why it thinks that's the intended destination.

**Reply by taholmes160** [reply]
Howdy -- heres the versions I am using:

```PS C:\Users\tahol> git version
git version 2.45.1.windows.1
PS C:\Users\tahol> git lfs version
git-lfs/3.5.1 (GitHub; windows amd64; go 1.21.7; git e237bb3a)
PS C:\Users\tahol>

```

I have been using git and git lfs for a couple years now -- backing up binary files from blender projects etc.  

my software HDD failed, so I had to uninstall all of my software, and re-install it -- thats when this problem started.  so Im not really sure what I should be doing b/c ive never had to mess with it before

thanks
TIM

**Reply by taholmes160** [reply]
I did also just find that I can upload to git-hub using LFS without a problem -- so im not sure what that means -- is something goofed on my local git server?  its a Ubuntu Linux server running 22.10 --

**Reply by bk2204** [reply]
You'd probably want to do `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1 git push origin main` and see what the output says.  You can attach it here as a text file (not an image) and we can look at the output to see why it's not working.

You could also try doing `ssh 192.168.1.28 git-lfs-transfer /mnt/data1/git/vivos.git download` and see if there's an error message printed.

**Reply by taholmes160** [reply]
ok, here we go -- I attached the output from the Git trace command 
[git trace.txt](https://github.com/git-lfs/git-lfs/files/15439128/git.trace.txt)

note the repo name has changed from vivos (My actual working project) to vivostest (a copy that I can afford to loose if I do something wrong)

Here is the output of the git-lfs-transfer command:

My server is set up for key auth, so it doesnt accept a password on ssh anymore --the git-lfs-transfer wants a password

Note that up until this problem started, I have not had problems with LFS and the key auth -- its been running that way for some time

**Reply by bk2204** [reply]
Okay, can you try running `ssh taholmes160@192.168.1.28 git-lfs-transfer /mnt/data1/git/vivostest.git upload` to see what the output is there?
